docs/pages/contributing/instantiations.md

```diff
@@ -0,0 +1,66 @@
+# Instantiating Classes in SparseBase
+
+While SparseBase supports a header-only mode, our main focus is in the compiled version of the library. 
+This is mainly due to the large impact of header-only mode on the build times of user's projects.
+
+As a result of this, we try to keep the header files as light as possible by avoiding putting implementations there.
+There are a few exceptions to this (like the `As<>()` function of `Format`) 
+but this is only done in very rare situations.
+
+Hence, any templated class or function you create in SparseBase will require instantiation as it needs to be compiled
+with the library with the types known at compile time. This is achieved with the script 
+`src/generate_explicit_instantiations.py`.
+
+> Note that this script is not intended to be called manually besides debugging. It will be called automatically
+> by CMake during the configuration step
+
+To add a new instantiation you need to call the following function. Each parameter is explained in its own section.
+
+```python
+def gen_inst(template, filename, ifdef=None, folder=output_folder):
+```
+
+## Template
+
+**Example:** `CSR<$id_type, $nnz_type, $value_type>`
+
+The following operations will be performed on the template you provide:
+
+1. Template will be placed in `template class <your_template>;\n`
+2. The identifiers starting with `$` will be replaced with their counterparts in the CMake variables. Currently, we have
+`$id_type`, `$nnz_type`, `$value_type`, `$float_type`. Every possible combination will be covered.
+
+
+## Filename
+
+The filename to be used for the output. As per [Google's Style Guide](https://google.github.io/styleguide/cppguide.html),
+the filenames we use end with the `.inc` extension. 
+
+> Caution!!! If the file is being used for the first time you should also call the `reset_file()` function with it.
```

Comment:
Same with above. Could be my doxygen idk

Reply:
Not related with doxygen. Read The Docs theme just defines them like that. Sphinx has dedicated warning and info blocks I will use them instead.

Reply:
Also this is not local to this page. For example our current live [getting started page](https://sparcityeu.github.io/sparsebase/pages/getting_started.html) has the same issue.